### PR TITLE
first version of web security

### PR DESCRIPTION
**Pull request by oleksiistepaniak**

### CODE REVIEW

**Review by AndrewkeY-hub — APPROVED** [review]

File: src/main/java/taxi/dao/CarDaoImpl.java [hunk 1]
```diff
@@ -211,10 +211,14 @@ private Driver parseDriverFromResultSet(ResultSet resultSet) throws SQLException
...
         String licenseNumber = resultSet.getString("license_number");
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
         Driver driver = new Driver();
         driver.setId(driverId);
         driver.setName(name);
         driver.setLicenseNumber(licenseNumber);
+        driver.setLogin(login);
+        driver.setPassword(password);
         return driver;
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/dao/CarDaoImpl.java:222]: please add selecting these fields in query as well

File: src/main/java/taxi/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
```
Comment (KseniiaMakarova) [inline src/main/java/taxi/web/filter/AuthenticationFilter.java:29]: bad naming

File: src/main/webapp/WEB-INF/views/drivers/all.jsp [hunk 3]
```diff
@@ -8,12 +8,15 @@
...
 <body>
+<%@include file="/WEB-INF/views/header.jsp"%>
 <h1 class="table_dark">All drivers:</h1>
 <table border="1" class="table_dark">
     <tr>
         <th>ID</th>
         <th>Name</th>
         <th>License number</th>
+        <th>Username</th>
+        <th>Password</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/drivers/all.jsp:19]: it is not correct to show password to everyone

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 4]
```diff
@@ -20,6 +21,7 @@
     <tr><td><a href="${pageContext.request.contextPath}/cars/add">Create new Car</a></td></tr>
     <tr><td><a href="${pageContext.request.contextPath}/manufacturers/add">Create new Manufacturer</a></td></tr>
     <tr><td><a href="${pageContext.request.contextPath}/cars/drivers/add">Add Driver to Car</a></td></tr>
+    <tr><td><a href="${pageContext.request.contextPath}/cars/drivers/all">Get my cars</a></td></tr>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/index.jsp:24]:
```suggestion
    <tr><td><a href="${pageContext.request.contextPath}/drivers/cars">Get my cars</a></td></tr>
```

File: src/main/webapp/web.xml [hunk 5]
```diff
@@ -105,4 +128,22 @@
...
         <url-pattern>/manufacturers/delete</url-pattern>
     </servlet-mapping>
+
+    <servlet>
+        <servlet-name>getAllCarsByDriver</servlet-name>
+        <servlet-class>taxi.controller.car.GetMyCurrentCarsController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>getAllCarsByDriver</servlet-name>
+        <url-pattern>/cars/drivers/all</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml:138]:
```suggestion
        <url-pattern>/drivers/cars</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
good job!

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**